database/triggers.py

```diff
@@ -0,0 +1,213 @@
+from alembic_utils.pg_function import PGFunction
+from alembic_utils.pg_trigger import PGTrigger
+
+hame_tables = [
```

Comment:
I'm thinking this can be done kinda reading from the existing models, so that if new hame tables are added, the triggers are added automatically. Perhaps this is too much magic and not clear enough though:

```
hame_tables = [klass.table_name for name, klass in inspect.getmembers(models, inspect.isclass) where issubclass(klass, models.VersionedBase)]

# I think also plan objects should have lifecycle dates added when their status is changed:
tables_with_lifecycle_date = [klass.table_name for name, klass in inspect.getmembers(models, inspect.isclass) where issubclass(klass, models.PlanBase)]

# plan should not be here, but all other lifecycle status tables should be:
tables_with_dependent_lifecycle_status = [klass.table_name for name, klass in inspect.getmembers(models, inspect.isclass) where issubclass(klass, models.PlanBase) and klass is not models.Plan]

plan_object_tables = [klass.table_name for name, klass in inspect.getmembers(models, inspect.isclass) where issubclass(klass, models.PlanObjectBase)]
```

Reply:
I think this is a great idea!

Reply:
@Rikuoja In our model, there is no relation between plan object tables and lifecycle_date table.

Reply:
@Rikuoja I noticed lifecycle_date was missing plan object ids so I added those.

Reply:
@msorvoja Good, I also actually am adding them in Ryhti validation Lambda fork, but probably you get your PR finished first, so I can remove them from my fork then.

Reply:
@msorvoja  Probably your fields are named the same as the ones I used: https://github.com/GispoCoding/hame-ryhti/blob/273-basic-ryhti-client/database/models.py#L410